activemodel/lib/active_model/errors.rb

```diff
@@ -238,6 +270,14 @@ def messages
       hash
     end
 
+    # Returns a Hash of attributes with an array of their custom error messages.
+    def custom_messages
```

Comment:
I'm wondering about use cases where I'm happy using the `full_message` for some attributes, but want to override it with a 'custom' full message for other attributes. I guess I'm imagining a `full_message` as a fallback, with the `custom_message` taking precedence if one can be found. That you'd only set the `custom_message` phrase where the combination of attribute and message won't work. How do you see `custom_messages` fitting into that kind of scenario? 🙇‍♂️

Reply:
Related to ☝️, if I don't set a `custom_message` translation for a particular error, then call `custom_messages`, what should happen? Should it error because it can't find a translation, or should it fall back to the `full_message`?

Reply:
That's an interesting scenario. Do you have a concrete example where this would be needed?

As I see it, `custom_message` is an alternative to displaying errors and as such, should probably have its own fallback chain like the current messages do. In the current implementation, it'd fall back to `errors.blank.custom_message`. 

If the purpose of it is to create an error with a friendlier, more on-brand message, I would prefer it to raise an error if a translation is not found. As an example, all copy in mu company has to be approved by someone with specific training. If a custom_message was missing, I would want an exception to be raised in CI.

Reply:
> Do you have a concrete example where this would be needed?

I'm thinking of a form where we show inline error messages next to each field, and list full sentence error messages at the top. 

In general, for the full sentence error messages we can use the `full_message` built by combining the attribute and message. 

So for a name field, the message can be "can't be blank" and the full sentence message can be "Name can't be blank".

But for one field - let's say phone number - we'd really like the full sentence message not to include the inline message. Suppose we want the inline message to be "is not a valid phone number" and the full sentence message to be "Phone number is invalid".

I'm imagining configuring the phrases like this:

```
en:
  activemodel:
    errors:
      models:
        user:
          attributes:
            name:
              blank: "can't be blank"
            phone_number:
              invalid: "is not a valid phone number"
              custom_messages:
                invalid: "Phone number is invalid"
```

Then calling `user.errors.custom_messages` (or `user.errors.full_messages`?) would return `["Name can't be blank", "Phone number is invalid"]`. 

Having to add "Name can't be blank" under `attributes.name.custom_messages.blank` (and having to add `custom_messages` phrases for all errors where the current `full_message` is fine) feels kind of inconvenient if I only want a truly _custom_ error message for particular errors...

However, maybe this is something to add within a project, along the lines of:
``` 
def full_sentence_errors
  errors.map do |error|
    error.custom_message || error.full_message
  end
end
```

Reply:
As you said, I believe this might be something you can do con a per-project or even per error basis. I imagine the following:

https://gist.github.com/pinzonjulian/7f291eba0a7229dbf1eac845553b049f